pkg/utils/tests/tests.go

```diff
@@ -68,3 +72,34 @@ func RequireSignal(t *testing.T, ch <-chan struct{}, failMsg string) {
 		t.Fatal(failMsg)
 	}
 }
+
+// Convenience method to setup a client connection with a default timeout of 5 seconds
+func SetupClientConnWithOptsAndTimeout(t *testing.T, target string, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
+	ctx, cancel := context.WithTimeout(Context(t), 5*time.Second)
+	t.Cleanup(cancel)
+
+	return grpc.DialContext(ctx, target, opts...)
+}
+
+// SetupServer is a convenience method to set up a grpc server for most testing
+// usecases.
+func SetupServer(t *testing.T, opts ...grpc.ServerOption) (net.Listener, *grpc.Server, string) {
+	// Attempt to reconnect to the port which the OS may not have had time
+	// to clean up between tests.
+	var (
+		lis net.Listener
+		err error
+	)
+
+	port := freeport.GetOne(t)
+	addr := fmt.Sprintf("localhost:%d", port)
+	require.Eventually(t, func() bool {
```

Comment:
i don't think the eventually is needed. it's a synchronous call that has no good reason to recover.

Reply:
The OS can still be freeing up the port while the next test is executed. This eventually block gives the OS some wiggle room to do that. This isn't strictly needed as you the current tests use freeport, but there may be scenarios in the future that want to use the same port across tests.

Reply:
freeport library handles the concern. there is no clear advantage to reusing ports across unit tests and there are clear disadvantages. we removed a number of flake integration test in core by adopting freeport instead of trying to manage it ourselves